R/py_require.R

```diff
@@ -0,0 +1,292 @@
+#' @export
+py_require <- function(packages = NULL,
+                       python_version = NULL,
+                       exclude_newer = NULL,
+                       action = c("add", "remove", "set")) {
+  action <- match.arg(action)
+
+  if (!is.null(exclude_newer) &&
+    action != "set" &&
+    !is.null(get_python_reqs("exclude_newer"))
+  ) {
+    stop(
+      "`exclude_newer` is already set to '",
+      get_python_reqs("exclude_newer"),
+      "', use `action` 'set' to override"
+    )
+  }
+
+  if (missing(packages) && missing(python_version) && missing(exclude_newer)) {
+    return(get_python_reqs())
+  }
+
+  remove_these <- function(items, from) {
```

Comment:
I think we can use `setdiff(pkgs, remove)` here. If we wanted to preserve `names()`, we could use `pkgs[!remove %in% pkgs]`.